File: app/api/results.py
from __future__ import annotations

from fastapi import APIRouter

from app.services.result_service import result_service

router = APIRouter(prefix="/api", tags=["results"])


@router.get("/tasks/{task_id}/result")
def get_task_result(task_id: str) -> dict:
    return result_service.get_result(task_id)


@router.get("/results")
def list_results() -> list[dict]:
    return result_service.list_results()

File: frontend/src/api/results.ts
import { apiClient, unwrap } from './client'; import type { SolveResult } from '../types/result';
export const getResults = () => unwrap<SolveResult[]>(apiClient.get('/api/results'));
export const getResult = (id: string) => unwrap<SolveResult>(apiClient.get(`/api/tasks/${id}/result`));
